Development

Underdog-01 opened the pull request:
This is a proposal for adding the ability to use < input > tags in the package-info.xml file.
It gives mod authors the ability to use HTML5 input and textarea for installing or uninstalling their modifications where the data can be processed in files accessed via the < code > tags.

The DCO.txt file has been manually signed to authorize the use of this code submission.

Please feel free to email me or leave me a message on my forum with any questions or feedback.
Thank you.

Commit:
Update Packages.php

Fill $context['new_inputs'] array with possible values from the form.

Commit:
Update Subs-Package.php

Source code for using < input > (<- no whitespaces) tags in package-info.xml to allow HTML5 input tag(s) to be used in the packages install/uninstall template. 
Please see the extra new_pkgs_commands-readme.txt file for the syntax to be used.

Commit:
Update Packages.template.php

This edit displays any custom inputs that were entered by the mod author using package-info.php

Commit:
Create new_pkgs_commands-readme.txt

This is the explanation of changes and the syntax for using < input > tags in the package-info.xml file. The data is passed via $context['new_inputs'] array to be processed by mod authors in any file used with < code > tags.

Underdog-01 commented:
This was tested on SMF 2.1 Alpha in which case it works as it should and without error.
Previous commits were deleted as they were branched to the SMF main repository in error. 
Also some on the fly file edits at the time made it look messy so I decided to clean it up.

Notes: Applying these edits as a modification would be inappropriate as it would involve users having to download this mod for developers to make use of it. Therefore adding this set of edits to one of the next releases would be more ideal. 

Without these edits, any author who would put < input > tags in their xml file would create a php error as your current code does not continue the loop for non existent xml tags which imo should be emended. 

@emanuele45 : Thank you for the instruction. I gave permission in the initial comment, however I will create an alias via my github terminal for future pull requests. If I did that permission incorrectly I will edit & do it properly once I figure out the exact procedure as per those instructions.  

It seems as though I will have to redo the instructions for the syntax as I somehow lost the updated one that was more legible. I will do that tomorrow when I have the time.

Commit:
Update new_pkgs_commands-readme.txt

Changes: 
- tabbed out text to make it more legible.

Commit:
Update new_pkgs_commands-readme.txt

Commit:
Update DCO.txt

Commit:
Create .gitconfig

matthew-kerle commented:
The biggest problem I see with this right now is that I've not yet committed HTML5 support to SMF 2.1, meaning that SMF 2.1 isn't running HTML5, yet this branch would allow HTML5 inputs.

Underdog-01 commented:
@Labradoodle-360 : It supports some of the newer HTML5 attributes but HTML5 would not be absolutely necessary to use this feature. Mod authors can be informed they should opt to only use the HTML4.1 standards even though the HTML5 ones are available. Imo perhaps HTML4.1 will ignore the HTML5 attributes that are within the < input > brackets when processed anyhow.

  The ones marked as NEW in the link shown below are HTML5 only and the rest are HTML4.1, as I said mod authors can be stressed to use the HTML4.1 standards. Not all attributes were implemented as they were not suitable for the packages template they are being used for. 
ref. http://www.w3schools.com/tags/tag_input.asp

  There is also the option of changing all the templates to HTML 5. All the major browsers support it & the DOCTYPE declaration is a much shorter syntax.

Underdog-01 committed:
Update new_pkgs_commands-readme.txt

corrections

Underdog-01 committed:
Update Subs-Package.php

matthew-kerle commented:
All I was saying is that HTML5 isn't supported by SMF 2.1 yet...I know all about HTML5.

live627 commented on new_pkgs_commands-readme.txt:
```diff
@@ -0,0 +1,103 @@
+number, password, radio, range, reset, search, submit, tel, text, textarea, time, url, week
+
+
+Allowed input/textare attributes:
```
typo

ghost commented:
ok besides for the typo that's easy fixed are we putting this in?

Oldiesmann commented:
The biggest issue here is that this is being submitted to the development branch, not the release-2.1 branch. If we're going to pull this in, it needs to be submitted to the release-2.1 branch instead.

emanuele45 commented on DCO.txt:
```diff
@@ -22,4 +22,8 @@ Developer's Certificate of Origin 1.1
+
+Signed,
+Underdog @ askusaquestion.net     
+Underdog-01 @ github.com            
```
These three lines should not be here.

ghost commented:
could someone else submit if Underdog doesnt reply?  But still credit underdog?

Underdog-01 commented:
My apologies ladies & gents. I was busy fixing up that Battle mod & did not notice your comments until today.
I will submit this to the release branch.
I am glad to see you like the idea.

Underdog-01 committed:
Update Packages.php

Underdog-01 commented:
Changes committed to Release2.1 branch... done.
I fixed the typos, changed some if/else to switch/case and changed some comparisons to typesafe. 
I'll test it again later today to make sure the switch/case is working properly. 
It does the same thing as if/else I suppose but Imo looks neater (aesthetically).

Arantor commented:
Closing as per https://github.com/SimpleMachines/SMF2.1/pull/579
